src/components/settings/GooglePermissions.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+import React, { Component } from 'react';
+import GoogleLogin from 'react-google-login';
+import { connect } from 'react-redux';
+import { flashErrorMessage } from 'redux-flash';
+import type { AppState } from '../../state/AppState';
+import type { User } from '../../models/User';
+import {
+  connectGoogle, allowDriveAccess, refreshUser, revokeDriveAccess,
+} from '../../state/auth/auth.actionCreators';
+import googleDriveBadge from '../../assets/googleDrive.png';
+import './GooglePermissions.scss';
+
+type Props = {
+  user: User,
+  dispatch: Function,
+}
+
+type State = {
+  googleAuth: any,
+}
+
+class GooglePermissions extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      googleAuth: null,
+    };
+  }
+
+  componentWillMount(): void {
+    const { dispatch, user } = this.props;
+    dispatch(refreshUser(user.id));
+    require('google-client-api')().then((gapi) => {
+      gapi.client.init({
+        clientId: process.env.REACT_APP_CLIENT_ID,
+        scope: 'profile email',
+      }).then(() => {
```

Comment:
This seems repeated with the logout. 
Maybe extract the gapi initialization to a separate file.
```
const gapi = await initGoogleAPIClient();
const googleAuth = gapi.auth2.getAuthInstance();
this.setState({ googleAuth });
```

Reply:
Or even 
```
const googleAuth = await getGoogleAuthInstance();
... 
```